// This file is part of mathtrader.
//
// Copyright (C) 2021 George Ioannidis
//
// mathtrader is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// mathtrader is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with mathtrader. If not, see <http://www.gnu.org/licenses/>.

// Proto definition for the mathtrade input, i.e., the result of parsing the
// Online Want List Generator (OLWLG) file.

syntax = "proto2";

package mathtrader;

import "mathtrader/common/wantlist.proto";

// Message encapsulating the trade input.
message ParserResult {
  // The wantlists generated by OLWLG.
  repeated Wantlist wantlist = 1;

  // Users with trading items.
  repeated string users = 2;

  // The number of items with an official name. This treats copies of individual
  // items as unique items. If no official names have been given, it corresponds
  // to the number of wantlists.
  optional int32 item_count = 3;

  // Tracks the items that have been declared as wanted items but have no
  // official name. Each item maps to its occurence across all wantlists.
  map<string, int32> missing_items = 4;
}
